### DISCUSSION TITLE
S3 data export

### DISCUSSION THREAD

**Discussion by influence2006** [Security-Onion-Solutions/securityonion#6014] [Unsupported Versions] [answered]
SO 2.3.80
Distributed setup.

I am trying to do 90-day log retention and after 90 days I want to send the old data to an Amazon S3 bucket.  I was wondering if anyone has any documentation on the best way to proceed with this?

**Comment by weslambert** [comment, marked as answer]
You might consider using Curator or using the Logstash s3 output plugin to send to s3.  Here is an example for Curator, however I've not tested it:

https://medium.com/@keniya.bhavya/elasticsearch-version-5-5-2-backup-restore-indices-from-s3-using-curator-a9316e2ea547

You may also try searching the rest of the Discussions for other threads with regard to backup/restore.

**Comment by BishalLakha** [comment]
@influence2006 were you able to send both data and indices using curator?

**Comment by influence2006** [comment]
They are adding the aws support to elasticsearch in 2.3.100. I had a setup
that was working but any time I had to restart a module I had to redo all
the setup. Figured I'll wait till there is aws integration and script that
is needed.

**Reply by BishalLakha** [reply]
We have distributed the architecture of SO hosted in AWS with a manager node and search node. I was able to backup snapshot for the manager node through kibana. But while trying to backup my search node following the article provided by @weslambert, however I get this error while creating a repository even when I use a valid access key. Do you have any idea about it?
 And can you share how you set this up at your end? Thank you.
![bug](https://user-images.githubusercontent.com/9823258/148657269-ba25dffd-a8d7-413d-bd36-24adfd33813c.png)

**Reply by influence2006** [reply]
I haven't seen this myself but here is an article on it. 
https://aws.amazon.com/premiumsupport/knowledge-center/s3-access-key-error/
